crates/katana/node/src/lib.rs

```diff
@@ -151,6 +153,17 @@ impl Node {
         let node_components = (pool, backend, block_producer, validator, self.forked_client.take());
         let rpc = spawn(node_components, self.rpc_config.clone()).await?;
 
+        // --- build and start the gas oracle worker task
+
+        // if the Option<GasWorkerConfig> is none, default to no sampling
+        if let Some(gpo_url) = self.l1_provider_url.as_ref() {
+            let gas_oracle = L1GasOracle::sampled(Some(gpo_url.clone())); // Clone the URL if needed
+            gas_oracle.run_worker(self.task_manager.task_spawner());
+        } else {
```

Comment:
You're still creating a separate instance of the gas oracle. The worker should be created based on the `L1GasOracle` instance that we pass to the `Backend` struct in the `.build` method. Hence, why the gas prices are not being updated.

Reply:
I see, so from what I understand I should do something like this? 
```rust
// in launch   self.backend.gas_oracle.run_worker(self.task_manager.task_spawner());
```